File: 283. 移动零/283. 移动零.go
package main

import "fmt"

func main() {
    // fmt.Println()
}

func moveZeroes(nums []int)  {
    cnt := 0
    for i := 0; i < len(nums); i++ {
        if nums[i] != 0 {
            nums[cnt] = nums[i]
            cnt += 1
        }
    }
    for j := 0; j < len(nums)-cnt; j++ {
        nums[j] = 0
    }


}

File: 283. 移动零/283. 移动零.py
class Solution:
    def moveZeroes(self, nums) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        # 没有要求空间复杂度O(1)，但是讲道理in-place就是要O(1)
        # 按顺序储存非0元素，修改，最后补0
        t=[]
        for each in nums:
            if each != 0:
                t.append(each)
        
        for i in range(len(t)):
            nums[i] = t[i]
        for i in range(len(t),len(nums)):
            nums[i] = 0
            
        





